sdcm/cluster.py

```diff
@@ -2939,7 +2952,31 @@ def node_setup(_node):
                 cl_inst.log.error(msg)
                 raise NodeSetupTimeout(msg)
 
-        time_elapsed = time.time() - start_time
+        start_time = time.perf_counter()
+        init_nodes = []
+        results = []
+
+        for node in node_list:
+            if isinstance(cl_inst, BaseScyllaCluster) and not Setup.USE_LEGACY_CLUSTER_INIT:
+                init_nodes.append(node)
+                if node.is_scylla_installed():
+                    node.stop_scylla()
+                    node.clean_scylla_data()
+                start_time = time.perf_counter()
+                node_setup(node)
+                verify_node_setup(start_time)
+                cl_inst.wait_for_nodes_up_and_normal(nodes=init_nodes, verification_node=node)
```

Comment:
This check is performed using `nodetool`.
For some reason, `nodetool` is very inefficient. It has a huge lag before it returns a result.

I have a suggestion: instead of querying nodetool, we could look at Scylla's log and wait until the "Scylla.*initialized" string appears. This should save us some time on bootstrapping.

Reply:
@bentsi this should be implemented in this pr or better as different task and PR?

Reply:
let's think about it more (open a task in Trello)